taskmgr/queueservice.go

```diff
@@ -53,22 +44,34 @@ type (
 		TaskClaimResponse   tcqueue.TaskClaimResponse      `json:"-"`
 		TaskReclaimResponse tcqueue.TaskReclaimResponse    `json:"-"`
 		Definition          tcqueue.TaskDefinitionResponse `json:"-"`
-		reclaimTimer        *time.Timer
 	}
 
-	TaskQueue struct {
+	taskQueue struct {
 		SignedDeleteUrl string `json:"signedDeleteUrl"`
 		SignedPollUrl   string `json:"signedPollUrl"`
 	}
 
-	TaskRuns []*TaskRun
+	taskRuns []*TaskRun
+
+	// TaskclusterQueue is an interface to the Queue client provided by the
+	// taskcluster-client-go package.  Passing around an interface allows the
+	// queue client to be mocked
+	TaskclusterQueue interface {
```

Comment:
Maybe we could call this simply `Queue` as it is in a different package to `github.com/taskcluster/taskcluster-client-id/queue` (i.e. from client side, it would be always be referred to as `taskmgr.Queue` rather than `queue.Queue` so the names would not collide). This might be nicer than `taskmgr.TaskclusterQueue`. Note, [this blog post](http://blog.golang.org/package-names) on package/type naming totally inspired me.